int32 mode
bool isCheckShowWrite
bool isCheckShowRead
int32[] data